pkg/model/components/kubelet.go

```diff
@@ -59,11 +59,27 @@ func (b *KubeletOptionsBuilder) BuildOptions(o interface{}) error {
 	// Standard options
 	clusterSpec.Kubelet.EnableDebuggingHandlers = fi.Bool(true)
 	clusterSpec.Kubelet.PodManifestPath = "/etc/kubernetes/manifests"
-	clusterSpec.Kubelet.AllowPrivileged = fi.Bool(true)
 	clusterSpec.Kubelet.LogLevel = fi.Int32(2)
 	clusterSpec.Kubelet.ClusterDomain = clusterSpec.ClusterDNSDomain
 	clusterSpec.Kubelet.NonMasqueradeCIDR = clusterSpec.NonMasqueradeCIDR
 
+	// AllowPrivileged is deprecated and removed in v1.14.
+	// See https://github.com/kubernetes/kubernetes/pull/71835
+	if kubernetesVersion.Major == 1 && kubernetesVersion.Minor < 14 {
+		if clusterSpec.Kubelet == nil {
```

Comment:
Nit: harmless, but we've already have panic-ed if Kubelet was nil :-)

Reply:
Sweet, cleaned up :)